src/Symfony/Component/Messenger/Bridge/Amqp/Tests/Transport/AmqpExtIntegrationTest.php

```diff
@@ -77,6 +77,36 @@ public function testItSendsAndReceivesMessages()
         $this->assertEmpty(iterator_to_array($receiver->get()));
     }
 
+    public function testItSendsAndReceivesMessagesThroughDefaultExchange()
+    {
+        $serializer = $this->createSerializer();
+
+        $connection = Connection::fromDsn(getenv('MESSENGER_AMQP_DSN'), ['exchange' => ['name' => '']]);
+        $connection->setup();
+        $connection->purgeQueues();
+
+        $sender = new AmqpSender($connection, $serializer);
+        $receiver = new AmqpReceiver($connection, $serializer);
+
+        $sender->send($first = new Envelope(new DummyMessage('First'), [new AmqpStamp('messages')]));
+        $sender->send($second = new Envelope(new DummyMessage('Second'), [new AmqpStamp('messages')]));
+
+        $envelopes = iterator_to_array($receiver->get());
+        $this->assertCount(1, $envelopes);
+        /** @var Envelope $envelope */
+        $envelope = $envelopes[0];
+        $this->assertEquals($first->getMessage(), $envelope->getMessage());
+        $this->assertInstanceOf(AmqpReceivedStamp::class, $envelope->last(AmqpReceivedStamp::class));
+
+        $envelopes = iterator_to_array($receiver->get());
+        $this->assertCount(1, $envelopes);
+        /** @var Envelope $envelope */
+        $envelope = $envelopes[0];
+        $this->assertEquals($second->getMessage(), $envelope->getMessage());
```

Comment:
```suggestion
        $this->assertSame($second->getMessage(), $envelope->getMessage());
```
whenever possible please, thanks

Reply:
@OskarStark thanks for pointing this out but unfortunately I cannot do that here because it leads to an error:
```
Failed asserting that two variables reference the same object.
```
And in this case, I believe it's perfectly fine to use `assertEquals` because it's suitable for objects comparison:
![image](https://github.com/user-attachments/assets/f84e80b6-c3d5-42ae-af89-71237370970f)